src/main/java/seedu/address/logic/commands/EditEventCommand.java

```diff
@@ -0,0 +1,195 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EVENT_TITLE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PURPOSE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_START_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_START_TIME;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.event.Event;
+import seedu.address.model.event.StartDate;
+import seedu.address.model.event.StartTime;
+
+/**
+ * Edits the details of an existing event in the application.
+ */
+public class EditEventCommand extends Command {
+
+    public static final String COMMAND_WORD = "editEvent";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the event identified "
+            + "by the index number used in the displayed event list. "
+            + "Existing values will be overwritten by the input values. \n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_EVENT_TITLE + "EVENT_TITLE]"
+            + "[" + PREFIX_START_DATE + "DATE]"
+            + "[" + PREFIX_START_TIME + "TIME]"
+            + "[" + PREFIX_PURPOSE + "PURPOSE] \n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_EVENT_TITLE + "Banana Sale "
+            + PREFIX_PURPOSE + "20% discount on all bananas store-wide";
+
+    public static final String MESSAGE_EDIT_EVENT_SUCCESS = "Edited Event: %1$s";
+
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    public static final String MESSAGE_DUPLICATE_EVENT = "This event already exists in the address book.";
+
+    private final Index index;
+
+    private final EditEventDescriptor editEventDescriptor;
+
+    /**
+     * Constructor of the EditEventCommand class
+     * @param index of the event to be edited in the filtered event list
+     * @param editEventDescriptor details to edit the event with
```

Comment:
May require full stops.
```suggestion
     * Constructor of the EditEventCommand class.
     * @param index of the event to be edited in the filtered event list.
     * @param editEventDescriptor details to edit the event with.
```